superset/viz.py

```diff
@@ -380,12 +380,6 @@ def get_df_payload(self, query_obj=None):
         if query_obj and not is_loaded:
             try:
                 df = self.get_df(query_obj)
-                if hasattr(self.datasource.database, 'db_engine_spec'):
-                    db_engine_spec = self.datasource.database.db_engine_spec
-                    df = db_engine_spec.adjust_df_column_names(df, self.form_data)
-                if self.status != utils.QueryStatus.FAILED:
-                    stats_logger.incr('loaded_from_source')
-                    is_loaded = True
```

Comment:
Same here: why is this broken? I'm worried changing this will break existing code.

Reply:
Removing this looks dangerous, the `hasattr` condition seems to be for preventing druid-related exception (a condition on that would be more explicit here). I'm pretty sure it's there for a reason.

Reply:
This is throwing a AttributeError on runtime saying `self.datasource.database` doesn't exist. Same as my comment before more than willing to add this functionality back but currently its not working at all.